src/game_inventory.cpp

```diff
@@ -1004,6 +1004,35 @@ class read_inventory_preset: public pickup_inventory_preset
             };
         }
 
+        bool sort_compare( const inventory_entry &lhs, const inventory_entry &rhs ) const override {
+            const bool base_sort = inventory_selector_preset::sort_compare( lhs, rhs );
+
+            const bool known_a = is_known( lhs.any_item() );
+            const bool known_b = is_known( rhs.any_item() );
+
+            if( !known_a || !known_b ) {
+                return ( !known_a && !known_b ) ? base_sort : !known_a;
+            }
+
+            const auto &book_a = get_book( lhs.any_item() );
+            const auto &book_b = get_book( rhs.any_item() );
+
+            if( !book_a.skill && !book_b.skill ) {
+                return ( book_a.fun == book_b.fun ) ? base_sort : book_a.fun > book_b.fun;
+            } else if( !book_a.skill || !book_a.skill ) {
+                return book_a.skill ? true : false;
```

Comment:
```suggestion
                return book_a.skill.has_value();
```

Reply:
Hmm, doesn't have has_value() which seems to be a cata::optional thing but i get back a string_id<>.

Possible alternatives:
!book_a.skill.is_null()
!!book_a.skill

Reply:
Stupid null ids. Use `!book_a.skill.is_null()` in that case.

Reply:
I settled for static_cast<bool>() which uses the skill_id operator just in case.